Src/VsVimShared/Implementation/Roslyn/RoslynListenerFactory.cs

```diff
@@ -78,23 +84,30 @@ private void OnIsRenameActiveChanged(object sender, EventArgs e)
                 _inRename = true;
                 foreach (var vimBuffer in _vimBufferList)
                 {
-                    vimBuffer.SwitchMode(ModeKind.ExternalEdit, ModeArgument.None);
+                    // Respect the user's edit monitoring setting.
+                    if (_vimApplicationSettings.EnableExternalEditMonitoring)
+                    {
+                        vimBuffer.SwitchMode(ModeKind.ExternalEdit, ModeArgument.None);
+                    }
                     vimBuffer.SwitchedMode += OnModeChange;
                 }
             }
         }
 
-        private void OnModeChange(object sender, EventArgs e)
+        private void OnModeChange(object sender, SwitchModeEventArgs args)
         {
-            if (_inRename && _roslynRenameUtil.IsRenameActive)
+            if (_inRename && _roslynRenameUtil.IsRenameActive && args.ModeArgument.IsCancelOperation)
             {
                 _roslynRenameUtil.Cancel();
             }
         }
 
         internal static bool IsRoslynContentType(IContentType contentType)

```

Comment:
Roslyn is generally C# / VB ony. Now that we are tracking F# as well should probably use a diff name. Suggest: `IsTargetContentType`

Reply:
But the problem then is much bigger: everything with RoslynRename in its name is named incorrectly. There are about half a dozen files and classes that will need to be renamed. Should I fix those as well? Example: RoslynRenameUtil.cs => InlineRenameUtil.cs.

Reply:
I renamed all of "RoslynRename" to "InlineRename" as that is what the class is called within Visual Studio.